ntc_rosetta/translators/openconfig/ios/openconfig_system/system.py

```diff
@@ -0,0 +1,229 @@
+            pass
+
+
+class Dns(Translator):
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/dns"
+
+    config = DnsConfig
+    servers = DnsServers
+
+
+class NtpServerConfig(Translator):
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/ntp/servers/server/config"
+
+
+class NtpServer(Translator):
+    config = NtpServerConfig
+
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/ntp/servers/server"
+
+        def pre_process_list(self) -> None:
+            if self.to_remove:
+                for element in self.to_remove:
+                    self.result.add_command(f"no ntp server {element['address']}")
+
+    def address(self, value: str):
+        self.yy.result.add_command(f"ntp server {value}")
+
+
+class NtpServers(Translator):
+    server = NtpServer
+
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/ntp/servers"
+
+
+class NtpConfig(Translator):
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/ntp/config"
+
+
+class Ntp(Translator):
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/ntp"
+
+    config = NtpConfig
+    servers = NtpServers
+
+
+class SshServerConfig(Translator):
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/ssh-server/config"
+
+    def protocol_version(self, value: str) -> None:
+        if value == "V2":
+            self.yy.result.add_command(f"ip ssh version 2")
+        elif value == "V1":
+            self.yy.result.add_command(f"ip ssh version 1")
+        else:
+            self.yy.result.add_command(f"default ip ssh version")
+
+    def timeout(self, value: int) -> None:
+        if value:
+            self.yy.result.add_command(f"ip ssh time-out {value}")
+
+
+class SshServer(Translator):
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/ssh-server"
+
+    config = SshServerConfig
+
+
+class AaaAuthenticationUserConfig(Translator):
+    class Yangify(TranslatorData):
+        path = "/openconfig-system:system/aaa/authentication/users/user/config"
+
+        def pre_process(self) -> None:
+            self.extra = {
+                "username": "",
+                "password": "",
+                "password_hash": "",
+                "role": "",
+                "ssh_key": "",
+            }
+
+        def post_process(self) -> None:
+            if not self.extra.get("username"):
+                return
+            role = self.extra.get("role")
+            role_str = f" privilege {role}" if role else ""
+            password = self.extra.get("password")
+            password_str = f"password 0 {password}" if password else ""
+            password_hash = self.extra.get("password_hash")
+            match = re.match(r"^\$(\d).*", password_hash)
+            password_hash_str = ""
+            if match:
+                hash_type = match.groups()[0]
```

Comment:
I think this might need to be adjusted to some kind of mapping or `if else`. It appears MD5 hashed passwords show as `$1` instead of a `$5` that would be needed for the command to succeed. It appears that md5 might be the exception to this so maybe an `if hash_type == 1: hash_type = 5` kind of thing?

Reply:
Could you elaborate on this a little more, maybe with a config example?  The match group grabs the digit from the hash that indicates type, but I'm not sure what you are saying to do with it based on the value.

Reply:
Yeah definitely. Here is an example from the DevNet always on IOS-XE box:
```
username test_md5 secret 5 $1$pdQG$blOvq4Ey0SJCv3n001vnj.
username test_type8 secret 8 $8$FiJQlYEnyjCvkk$GkMYqqmLNP2YNoAFBcod4Tg7/Sy6e58Q3a8umokmlJI
username test_type9 secret 9 $9$8OnHa3BCA8nvl.$guLZp0qgTdyTma10Tk.O7iUO5NufJhOnDoUcvy1R79I

```
From my understanding of the code it will translate the hashed passwords and determine the hash_type after the first **$**. The issue is with the test_md5 or any MD5 password,, it will use translate to the following command: `username test_md5 secret 1 $1$pdQG$blOvq4Ey0SJCv3n001vnj.`, but will error out: `Invalid encryption type: 1.  Password not set.`
```
csr1000v-1(config)#username test secret ?
  0      Specifies an UNENCRYPTED secret will follow
  5      Specifies a MD5 HASHED secret will follow
  8      Specifies a PBKDF2 HASHED secret will follow
  9      Specifies a SCRYPT HASHED secret will follow
  <0-9>  Encryption types not explicitly specified
  LINE   The UNENCRYPTED (cleartext) user secret
  LINE   The UNENCRYPTED (cleartext) user secret
```
We need to account for that and if the `hash_type` is 1, it should be converted to 5 to properly work on Cisco.

Reply:
Ok, I see.  Thanks for pointing that out!